starknet_py/net/models/transaction.py

```diff
@@ -76,13 +87,69 @@ class AccountTransaction(Transaction, ABC):
 TypeAccountTransaction = TypeVar("TypeAccountTransaction", bound=AccountTransaction)
 
 
+@dataclass(frozen=True)
+class _AccountTransactionV3(AccountTransaction, ABC):
```

Comment:
Why do we start the class name with an underscore?
If it means it's some sort of internal class, I assume `AccountTransaction` should also be `_AccountTransaction`?

Reply:
I agree that there is some inconsistency. However, `AccountTransaction` is utilisied in `Client` methods for all types of transaction, while `_AccountTransactionV3` is a helper class added to reduce redundant code and I don't want to expose it to users. I propose either renaming `_AccountTransactionV3` or introducing another internal class `_DeprecatedAccountTransaction` or `_AccountTransactionV1` (not sure how to handle `v2` then, since  `_AccountTransactionV2` would be identical to `_AccountTransactionV1`).

Reply:
> I agree that there is some inconsistency. However, `AccountTransaction` is utilisied in `Client` methods for all types of transaction, while `_AccountTransactionV3` is a helper class added to reduce redundant code and I don't want to expose it to users.

Yes, the inconsistency is what alerts me a bit, I didn't find any other case where class is named that way. With that said, you argument makes sense. I don't know if the class is necessarily something that should be "hidden" from the users, so it's up to you. 

> introducing another internal class `_DeprecatedAccountTransaction` or `_AccountTransactionV1` (not sure how to handle `v2` then, since `_AccountTransactionV2` would be identical to `_AccountTransactionV1`).

`_DeprecatedAccountTransaction` should be good for both v1 and v2.